sdk/test-utils/perf/src/managerProgram.ts

```diff
@@ -0,0 +1,300 @@
+    )) as ReportResultsMessage[];
+
+    // stop the handleUpdate part when it next gets a chance
+    done = true;
+
+    const results = resultMessages.map((m) => m.snapshots).flat();
+
+    console.log(`=== ${title} mode, results of iteration ${iterationIndex + 1} ===`);
+    this.logResults(results);
+
+    await stage;
+  }
+
+  private async logPackageVersions(listTransitiveDeps: boolean): Promise<void> {
+    return new Promise((resolve) => {
+      console.log("=== Versions ===");
+      exec(`npm list --prod ${listTransitiveDeps ? "" : "--depth=0"}`, (_error, stdout) => {
+        for (const dependency of stdout.split("\n").filter((line) => line.includes("@azure"))) {
+          console.log(dependency);
+        }
+        resolve();
+      });
+    });
+  }
+
+  /**
+   * The run() public method lets developers specify when to begin running the selected test
+   * under the conditions provided by the command line options.
+   * If the command line option for help (--help or -h) is passed in, the program will output
+   * the information available of all of the options and close, with no test executions.
+   *
+   * This method will invoke the test class's "globalSetup" and "globalCleanup" once per CPU,
+   * and the "setup" and "cleanup" as many times as tests were instantiated
+   * (up to the "parallel" command line parameter, which defaults to 1).
+   *
+   * If the "warmup" command line parameter is defined, the tests will be called
+   * until as many seconds as the "warmup" parameter says. This is to adjust
+   * to any possible real-time optimizations that the JavaScript runtime might
+   * do while executing something repeatedly. This is also a requirement to align with the
+   * Perf framework in other languages.
+   *
+   * If any exception is encountered, the whole process will stop, unless
+   * these exceptions are instances of the class PerfTestError, which defines expected errors.
+   */
+  public async run(): Promise<void> {
+    // There should be no test execution if the help option is passed.
+    // --help, or -h
+    if (this.parsedOptions.help.value) {
+      console.log(`=== Help: Options that can be sent to ${this.testName} ===`);
+      console.table(this.dummyTestInstance.parsedOptions);
+      return;
+    }
+
+    await this.logPackageVersions(
+      this.parsedOptions["list-transitive-dependencies"].value ?? false
+    );
+
+    const options = this.dummyTestInstance.parsedOptions;
+    console.log("=== Parsed options ===");
+    console.table(options);
+
+    this.createWorkers();
+
+    console.log(
+      `=== Calling globalSetup() once per CPU for (all) the instance(s) of ${this.testName} ===`
+    );
+    await performStage("globalSetup");
+
+    console.log(
+      `=== Calling setup() for the ${this.parallelNumber} instantiated ${this.testName} tests ===`
+    );
+
+    await performStage("setup");
+
+    await performStage("postSetup");
+
+    if (Number(options.warmup.value) > 0) {
```

Comment:
```suggestion
    if (options.warmup.value > 0) {
```
?

Reply:
huh, wonder why this is here. Guess it might be copied from the old code.